model.py

```diff
@@ -221,6 +179,157 @@ def forward(self, outputs, targets):
+    }
+
+    criterion = SetCriterion(
+        configuration.num_classes,
+        matcher=matcher,
+        weight_dict=weight_dict,
+        eos_coef=configuration.eos_coef,
+        losses=losses,
+    )
+    return criterion
+
+
+class DETR(pl.LightningModule):
+    """
+    This is the main model that performs segments detection and classification
+
+    Copy-paste from DETR module with modifications
+    """
+
+    def __init__(
+        self,
+        transformer,
+        num_classes,
+        num_queries,
+        num_nucleobase_letters,
+        criterion,
+        configuration,
+    ):
+        """Initializes the model.
+        Parameters:
+            transformer: torch module of the transformer architecture.
+            num_classes: number of object classes
+            num_queries: number of object queries, ie detection slot. This is the maximal number of objects
+                        DETR can detect in an subsequence.
+        """
+        super().__init__()
+        self.num_queries = num_queries
+        self.num_classes = num_classes
+        self.transformer = transformer
+        hidden_dim = transformer.d_model
+        self.emb = nn.Embedding(num_nucleobase_letters, hidden_dim)
+        self.class_embed = nn.Linear(hidden_dim, num_classes + 1)
+        self.segment_embed = MLP(hidden_dim, hidden_dim, 2, 3)
+        self.query_embed = nn.Embedding(num_queries, hidden_dim)
+        self.pe = PositionalEncoding(hidden_dim)
+        self.criterion = criterion
+        self.configuration = configuration
+
+    def forward(self, sample: Tensor, seq_starts: List[int]):
+        """
+        Parameters:
+            -- sample: batched sequences, of shape [batch_size x seq_len x embedding_len ]
+                 eg, when choosing one hot encoding, embedding_len will be 5, [A, T, C, G, N]
+        Return values
+            -- pred_logits: the classification logits (including no-object) for all queries.
+                            Shape= [batch_size x num_queries x (num_classes + 1)]
+            -- pred_boundaries: The normalized boundaries coordinates for all queries, represented as
+                            (center, width). These values are normalized in [0, 1].
+        """
+        sample = self.emb(sample)
+        pos = self.pe(sample)
+        hs = self.transformer(sample, self.query_embed.weight, pos)
+        outputs_class = self.class_embed(hs)
+        outputs_coord = self.segment_embed(hs).sigmoid()
+
+        # ([batch_size]) -> [batch_size, num_queries]
+
+        seq_starts = [
+            [seq_start for _ in range(self.num_queries)] for seq_start in seq_starts
+        ]
+        out = {
+            "pred_logits": outputs_class,
+            "pred_boundaries": outputs_coord,
+            "seq_start": seq_starts,
+        }
+        return out
+
+    def training_step(self, batch, batch_idx):
+        samples, seq_starts, targets = batch
+        targets = [{k: v for k, v in t.items()} for t in targets]
```

Comment:
```
        targets = [{k: v for k, v in t.items()} for t in targets]
```
What does this line do? It seems like it is not changing `targets` at all to me.

Reply:
The test result seems useless, and now I remove it.